### PR TITLE
Set up search-v2-operator-bundle repo

### PR DESCRIPTION
**Pull request by dislbenn**
Related issue: https://github.com/stolostron/backlog/issues/25294
Signed-off-by: dislbenn <dbennett@redhat.com>

### CODE REVIEW

File: Makefile.prow [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Copyright Contributors to the Open Cluster Management project
```
Comment (jlpadilla) [inline Makefile.prow]: Do we need this? There isn't any go code in this project.

File: bundle/manifests/search.open-cluster-management.io_searches.yaml [hunk 2]
```diff
@@ -0,0 +1,349 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (jlpadilla) [inline bundle/manifests/search.open-cluster-management.io_searches.yaml:1]: @xrajesh Some of these files get updated automatically. What's the best way to ensure that these stay in sync with code changes in the operator project?

File: sonar-project.properties [hunk 3]
```diff
@@ -0,0 +1,13 @@
+###############################################################################
```
Comment (jlpadilla) [inline sonar-project.properties]: Do we require sonar? There's no go code to test or scan in this project.

Comment (xrajesh) [reply sonar-project.properties]: We dont need sonar.

File: Makefile.prow [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+.PHONY: lint
+lint:
+	GOPATH=$(go env GOPATH)
+	curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b "${GOPATH}/bin" v1.47.1
+	CGO_ENABLED=0 GOGC=25 golangci-lint run --timeout=3m
+	gosec ./...
+
+.PHONY: unit-test
+unit-test:
+	go test ./... -v -coverprofile cover.out
```
Comment (xrajesh) [inline Makefile.prow]: Makefile also does not need lint and unit-test targets , this need the script to populate the right images - and when we have the e2e test , a target for that.

File: sonar-project.properties (hunk 3 above)
Comment (dislbenn) [reply sonar-project.properties]: Removed the sonar file.

File: Makefile.prow [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+-include /opt/build-harness/Makefile.prow
+
+.PHONY: bundle-latest-image
+bundle-latest-image:
+	bash ./scripts/bundle-image-pickup.sh -s
+
+.PHONY: e2e
+e2e:
+	echo "Running e2e test..."
```
Comment (jlpadilla) [inline Makefile.prow]: Update message for transparency and help us debug later.
```suggestion
	echo "[TODO] Run e2e test here."
```

File: bundle/manifests/search.open-cluster-management.io_searches.yaml (hunk 2 above)
Comment (xrajesh) [reply bundle/manifests/search.open-cluster-management.io_searches.yaml:1]: @jlpadilla  If the operator is running with different version of CRD (1) Bundle install should fail if the versions mismatch - , when we apply the CR for the operator (2) If there is field changes that should be tested in our e2e for all the options.

File: scripts/bundle-image-pickup.sh [hunk 6]
```diff
@@ -0,0 +1,272 @@
...
+      echo -e "${PURPLE}$2 ${NC}"$3
+    ;;
+    red)
+      echo -e "${RED}$2 ${NC}"$3
+    ;;
+    yellow)
+      echo -e "${YELLOW}$2 ${NC}"$3
+    ;;
+  esac
+}
```
Comment (jlpadilla) [inline scripts/bundle-image-pickup.sh]: **Minor suggestion.** I'd move this to `scripts/log-colors.sh`, then import here with `source ./scripts/log-colors.sh`

Then it's easier to use in other scripts and sync with other projects like our e2e-test

File: scripts/bundle-image-pickup.sh [hunk 7]
```diff
@@ -0,0 +1,272 @@
...
+  DEFAULT_POSTGRESQL_SNAPSHOT=`cat ./pg-snapshot.ver`
+
+elif [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
+  log_color "red" "ERROR: Silent mode will not work when ./snapshot.ver is missing"
+  exit 1
+fi
+
+# Input or fetch the latest build image to update the csv file.
+if [[ " $@ " =~ " --silent " || " $@ " =~ " -s " ]]; then
+  log_color "cyan" "** Running in silent mode **"
```
Comment (jlpadilla) [inline scripts/bundle-image-pickup.sh:165]: This is nice, please document the usage with a comment at the top of this script.

File: Makefile.prow (hunk 4 above)
Comment (dislbenn) [reply Makefile.prow]: Removed unit-test and lint targets.

File: scripts/bundle-image-pickup.sh (hunk 6 above)
Comment (dislbenn) [reply scripts/bundle-image-pickup.sh]: Added `log_color` to `scripts/log-colors.sh`. Great suggestion!

File: Makefile.prow (hunk 5 above)
Comment (dislbenn) [reply Makefile.prow]: Added suggestion to `Makefile`.

**Review by jlpadilla — APPROVED** [review]
/lgtm
/hold

@dislbenn This is a great start to this repo. We can merge to unblock the rest of the pipeline work and handle any improvements in a separate PR.

**Comment by dislbenn** [comment]
@jlpadilla merging this PR.

**Pull request merged**